File: Java/exercise56_spring/src/main/java/com/murilomartins/services/SalaryService.java
package com.murilomartins.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.murilomartins.entities.Employee;

@Service
public class SalaryService {

	@Autowired
	private TaxService taxService;
	@Autowired
	private PensionService pensionService;
	
	public double netSalary(Employee emp) {	
		return emp.getGrossSalary() - taxService.tax(emp.getGrossSalary()) - pensionService.discount(emp.getGrossSalary());
	}
}

File: JavaScript/salario_liquido/js/services/salaryService.js
import { tax } from "./taxService.js";
import * as pensionService from "./pensionService.js";

export function netSalary(employee) {
    const pension = pensionService.discount(employee.grossSalary);
    return employee.grossSalary - tax(employee.grossSalary) - pension;
}